presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionContext.java

```diff
@@ -173,6 +184,12 @@ else if (nameParts.size() == 2) {
         transactionId = parseTransactionId(transactionIdHeader);
 
         this.sessionFunctions = parseSessionFunctionHeader(servletRequest);
+        if (isTracingEnabled()) {
+            this.tracer = Optional.of(requireNonNull(tracer, "tracer is null"));
+        }
+        else {
+            this.tracer = Optional.of(NoopTracerProvider.NOOP_TRACER);
```

Comment:
May be better to add a function comment to clarify that the input tracer is only used when isTracingEnabled() is true. Otherwise, developer might have a wrong assumption that whatever passed in is used.

Reply:
+1

Reply:
Good point. Let me update.